apps/minifront/src/state/staking.ts

```diff
@@ -130,6 +130,13 @@ const byBalanceAndVotingPower = (valueViewA: ValueView, valueViewB: ValueView):
   return byVotingPower;
 };
 
+/**
+ * To optimize performance while streaming potentially hundreds of
+ * delegations/validators in rapid succession, we only "flush" to state at a
+ * fixed interval, so as to prevent React from having to re-render too often.
+ */
+const DELEGATIONS_FLUSH_TO_STATE_INTERVAL = 50;
```

Comment:
is 50 too high? It seems to be taking a while on my machine

https://github.com/penumbra-zone/web/assets/16624263/72f28bdc-713e-42f4-ac3d-3bf78bb33498

Reply:
~Interesting... I was only testing this against my local devnet, which was more or less "infinitely" fast since there was no load on the server and no network lag time. So I only needed to optimize for _rendering_ performance, rather than _network_ performance, which made `50` an optimal interval.~

~But then I connected to `grpc2` and saw results more similar to yours. So I guess it's a balance between optimizing for rendering vs. network performance. I lowered the interval just now to `5` and the results seem quite a bit better, but still way slower than when connected to devnet. Perhaps a future optimization would be to cache the results between loads and only update balances.~

Wait a sec, this is weird. Loading speeds should be the same regardless of which RPC is being used, since validators are cached locally in IndexedDB. Going to keep investigating...

Reply:
@grod220 Think I figured it out. Can you try checking out [`jessepinho/metadata-fix`](https://github.com/penumbra-zone/web/pull/719) and testing that? It will likely be slow again the first time, then much faster on subsequent loads.

It was performing slowly because it had to fetch asset metadata for each delegation token. On my branch above, it saves these metadatas to the database as it fetches them, so that subsequent loads are faster.

(If this works, please approve [the PR](https://github.com/penumbra-zone/web/pull/719) so I can merge that branch into this one.)